src/main/java/net/jodah/typetools/TypeResolver.java

```diff
@@ -324,14 +323,10 @@ private static void populateLambdaArgs(Class<?> functionalInterface, final Class
             break;
           }
 
-          int methodRefOffset = resolveMethodRefOffset(constantPool);
-          String[] methodRefInfo = constantPool.getMemberRefInfoAt(constantPool.getSize() - methodRefOffset);
-
-          // Skip auto boxing methods
-          methodRefOffset = resolveAutoboxedMethodRefOffset(constantPool);
-          if (methodRefOffset != -1 && methodRefInfo[1].equals("valueOf")
-              && constantPool.getSize() > MIN_CONSTANT_POOL_SIZE)
-            methodRefInfo = constantPool.getMemberRefInfoAt(constantPool.getSize() - methodRefOffset);
+          String[] methodRefInfo = getMethodRefInfo(constantPool);
+          if (methodRefInfo == null) {
```

Comment:
I'm trying to think of when we wouldn't be able to resolve methodRefInfo... ideally we're always working with a lambda if we get this far and the method ref should be in there.